jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -967,7 +936,7 @@ protected void doStop() throws Exception
             stopContext();
 
             // retain only durable listeners
-            setEventListeners(_durableListeners.toArray(new EventListener[_durableListeners.size()]));
+            setEventListeners(_durableListeners);
```

Comment:
Are there any tests that test restarts of the context contain the correct number of listeners?

Reply:
Yes in ServletLifeCycleTest.  Adding another...